src/commands/aksRetinaCapture/aksRetinaCapture.ts

```diff
@@ -0,0 +1,126 @@
+    const clusterExplorer = await k8s.extension.clusterExplorer.v1;
+
+    if (!kubectl.available) {
+        vscode.window.showWarningMessage(`Kubectl is unavailable.`);
+        return;
+    }
+
+    if (!cloudExplorer.available) {
+        vscode.window.showWarningMessage(`Cloud explorer is unavailable.`);
+        return;
+    }
+
+    if (!clusterExplorer.available) {
+        vscode.window.showWarningMessage(`Cluster explorer is unavailable.`);
+        return;
+    }
+
+    const clusterInfo = await getKubernetesClusterInfo(target, cloudExplorer, clusterExplorer);
+    if (failed(clusterInfo)) {
+        vscode.window.showErrorMessage(clusterInfo.error);
+        return;
+    }
+
+    const kubectlRetinaPath = await getRetinaBinaryPath();
+    if (failed(kubectlRetinaPath)) {
+        vscode.window.showWarningMessage(`kubectl retina path was not found ${kubectlRetinaPath.error}`);
+        return;
+    }
+
+    ensureDirectoryInPath(path.dirname(kubectlRetinaPath.result));
+
+    const extension = getExtension();
+    if (failed(extension)) {
+        vscode.window.showErrorMessage(extension.error);
+        return;
+    }
+
+    const kubeConfigFile = await tmpfile.createTempFile(clusterInfo.result.kubeconfigYaml, "yaml");
+    console.log(`Kubeconfig file: ${kubeConfigFile}`);
+
+    // Get all Nodes For this Cluster
+    const nodelistResult = await invokeKubectlCommand(
+        kubectl,
+        kubeConfigFile.filePath,
+        `get nodes -o jsonpath="{range .items[*]}{.metadata.name}{' '}{end}"`,
+    );
+
+    if (failed(nodelistResult)) {
+        vscode.window.showErrorMessage(`Failed to capture the cluster: ${nodelistResult.error}`);
+        return;
+    }
+    const nodesList = nodelistResult.result.stdout.trim().split(" ").map((nodeName) => ({ label: nodeName }));
+
+    // Pick a cluster from group to compare with
+    const nodeNamesSelected = await vscode.window.showQuickPick(nodesList, {
+        canPickMany: true,
+        placeHolder: "Please select all the Nodes you want Retina to capture traffic from.",
+        title: "Select Nodes to Capture Traffic From",
+    });
+
+    if (!nodeNamesSelected) {
+        vscode.window.showErrorMessage('No nodes were selected to capture traffic.');
+        return;
+    }
+
+    const selectedNodes = nodeNamesSelected.map((item) => item.label).join(",");
+
+    if (!selectedNodes) {
+        return;
+    }
+
+    // Retina Run Capture
+    const capturename = `retina-capture-${clusterInfo.result.name}`;
+    const retinaCaptureResult = await longRunning(`Retina Distributed Capture running for cluster ${clusterInfo.result.name}.`, async () => {
+        return await invokeKubectlCommand(
+            kubectl,
+            kubeConfigFile.filePath,
+            `retina capture create --namespace default --name ${capturename} --host-path /mnt/capture --node-selectors "kubernetes.io/os=linux" --node-names "${selectedNodes}" --no-wait=false`,
```

Comment:
Is there a plan to support user-selection of the namespace for this? And also duration, or any of the other capture options?

Might be worth specifying duration (this will capture for 1 minute as it stands, won't it?) in the long-running task text, so the user knows how long they have to wait.

Reply:
Thanks for this question, its an interesting there is not hard recommendation to use the `capture` namespaces in some sample I see it does and in few it doesn't so keep day-0 simple command on `default namepsace` currently retina vanilla command run under default and capture, **RE:** duration, Retina has the `--no-wait` flag which is best way to make sure there is retina capture complete 100% before we move forward  which is best way for assurance for rest of stuff to work. ☕️❤️🙏

For anything else we could add future PR as and we enhance this please. If for anything uber missed please feel free to add the PR, thank you for this.

Kind fyi to @hsubramanianaks just in case. (also @sprab if anyone else has any thoughts.)